### ISSUE TITLE
can chat-with-rendezvous work on two nodes both behind nat?

### ISSUE THREAD

**Issue by hak1985** [libp2p/go-libp2p-examples#71]
rt,and if can work, does it use some public relay node or direct connect?

**Comment by upperwal** [comment]
@hak1985 Yes, if UPnP is enabled and working. Hole punching is not yet implemented in libp2p.

It will always work with a relay node.
